core/common.js

```diff
@@ -72,3 +74,30 @@ const setParentContainer = function(newParent) {
   parentContainer = newParent;
 };
 exports.setParentContainer = setParentContainer;
+
+/**
+ * All of the connections on blocks that are currently being dragged.
+ * @type {!Array.<!Connection>}
+ */
+let draggingConnections = [];
+
+/**
+ * Get the list of dragging connections.
+ * @returns {!Array.<!Connection>} The list of connections on blocks that are
+ *     currently being dragged.
+ */
+const getDraggingConnections = function() {
+  return draggingConnections;
+};
+/** @package **/
+exports.getDraggingConnections = getDraggingConnections;
+
+/**
+ * Set the list of dragging connections.
+ * @param {!Array.<!Connection>} connectionList The new list of connections on
+ *     blocks that are currently being dragged.
+ */
+const setDraggingConnections = function(connectionList) {
+  draggingConnections = connectionList;
+};
+exports.setDraggingConnections = setDraggingConnections;
```

Comment:
This API is fine, but—noting that the styleguide does _not_ prohibit _modifying the properties of an exported object_ (only modifying `exports` itself)—a simpler alternative would be to export the array itself:

```
/**
 * All of the connections on blocks that are currently being dragged.
 * @type {!Array<!Connection>}
 */
exports.draggingConnections = [];
```

And then replace calls to `Blockly.common.setDraggingConnections([])` with a simple assignment `Blockly.common.draggingConnections.length = 0;`, and modify the code in `Blockly.BlockSvg.prototype.setDragging` to append, i.e., replace:

```JS
    var newList =
        Blockly.common.getDraggingConnections().concat(this.getConnections_(true));
    Blockly.common.setDraggingConnections(newList);
```

with:
```JS
    Blockly.common.draggingConnections.push(...this.getConenctions_(true));
```
This is marginally less encapsulated, but it's definitely less complex, and the reduction in number of function calls and array allocations might make it a bit faster too.

Reply:
Done.